### PR TITLE
Feature/audio-system

### PR DESCRIPTION
**Pull request by PiterGroot**
This PR contains implemented sounds for:
* Player join
* Pickup / Drop
* Coal deposit
* Button sounds
* Deliver sounds
* Repair sounds
* Background song
* Notification popup 
* Train buttons / switch
* Coal oven on / off

### CODE REVIEW

**Comment by PiterGroot** [comment]
~TODO: Add coal oven toggle state sound~

File: Assets/Scripts/Framework/CoalOvenSystem/CoalOven.cs [hunk 1]
```diff
@@ -129,5 +147,11 @@ private void OnHealthChanged(int health)
 
             TrainEngine.Instance.SetEngineState(TrainEngineState.Inactive);
         }
+
+        [ObserversRpc(RunLocally = true, BufferLast = true)]
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/CoalOvenSystem/CoalOven.cs]: `BufferLast` can be removed.

File: Assets/Scripts/Framework/CoalOvenSystem/CoalOven.cs [hunk 2]
```diff
@@ -129,5 +147,11 @@ private void OnHealthChanged(int health)
...
+        private void PlayOvenSound(bool ovenEnabled)
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/CoalOvenSystem/CoalOven.cs]:
```suggestion
        private void PlayOvenSound(bool isOvenEnabled)
```

File: Assets/Scripts/Framework/DamageRepairManagement/TrainDamageable.cs [hunk 3]
```diff
@@ -53,10 +54,15 @@ private protected virtual void UpdateTimer()
         public override void Repair()
         {
             base.Repair();
+            PlayRepairSound();
 
             p_damageIntervalElapsed = 0;
         }
 
+        [ObserversRpc(RunLocally = true, BufferLast = true)]
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/DamageRepairManagement/TrainDamageable.cs]: `BufferLast` can be removed.

File: Assets/Scripts/Framework/Gameplay/Interactions/Grabbables/Grabbable.cs [hunk 4]
```diff
@@ -115,13 +118,28 @@ public virtual void PlaceOnGround()
...
+            if (ObjectParentUtils.TryGetObjectParent(hit.collider.gameObject, out ObjectParent objectParent))
             {
                 objectParent.SetParent(NetworkObject);
+                PlayGrabbableSound(true);
                 return;
             }
 
             NetworkObject.UnsetParent();
+            PlayGrabbableSound(true);
+        }
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/Gameplay/Interactions/Grabbables/Grabbable.cs]: You can just have 1 `PlayGrabbableSound()` call above the ObjectParent if statement.

File: Assets/Scripts/Framework/Gameplay/Interactions/Grabbables/Grabbable.cs [hunk 5]
```diff
@@ -115,13 +118,28 @@ public virtual void PlaceOnGround()
...
+
+        [ObserversRpc(RunLocally = true, BufferLast = true)]
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/Gameplay/Interactions/Grabbables/Grabbable.cs]: `BufferLast` can be removed.

File: Assets/Scripts/Framework/Gameplay/Interactions/Interactables/CoalOvenInteractable.cs [hunk 6]
```diff
@@ -74,9 +75,15 @@ public override bool Interact(UseableGrabbable useableGrabbable)
             useableGrabbable.OriginInteractor.UpdateInteractingTarget(useableGrabbable.OriginInteractor.Owner, null, false);
             useableGrabbable.Despawn();
 
+            PlayDepositSound();
+
             return base.Interact(useableGrabbable);
         }
 
+        [ObserversRpc(RunLocally = true, BufferLast = true)]
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/Gameplay/Interactions/Interactables/CoalOvenInteractable.cs]: `BufferLast` can be removed.

File: Assets/Scripts/Framework/Gameplay/Interactions/Interactables/TrainSpeedButtonInteractable.cs [hunk 7]
```diff
@@ -2,6 +2,8 @@
 
 using DerailedDeliveries.Framework.Gameplay.Player;
 using DerailedDeliveries.Framework.Train;
+using FishNet.Object;
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/Gameplay/Interactions/Interactables/TrainSpeedButtonInteractable.cs]: Move up a bit.

File: Assets/Scripts/Framework/Gameplay/Interactions/Interactables/TrainSpeedButtonInteractable.cs [hunk 8]
```diff
@@ -26,8 +28,14 @@ private protected override bool Use(Interactor interactor)
             if(!base.Use(interactor))
                 return false;
 
+            PlayUseSound();
             TrainEngine.Instance.AdjustSpeed(_isForwardButton);
+
             return true;
         }
+
+        [ObserversRpc(RunLocally = true, BufferLast = true)]
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/Gameplay/Interactions/Interactables/TrainSpeedButtonInteractable.cs]: `BufferLast` can be removed.

File: Assets/Scripts/Framework/UI/TextUpdaters/Lobby/PlayerListTextUpdater.cs [hunk 9]
```diff
@@ -2,6 +2,7 @@
 using UnityEngine;
 
 using DerailedDeliveries.Framework.PlayerManagement;
+using DerailedDeliveries.Framework.Audio;
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/UI/TextUpdaters/Lobby/PlayerListTextUpdater.cs]:
```suggestion
```

File: Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs [hunk 10]
```diff
@@ -224,6 +226,9 @@ private void SpawnPlayerOnServer(NetworkConnection clientConnection)
             _availableSpawnpoints.RemoveAt(randomSpawnIndex);
 
             _playerIdCount++;
+
+            if (StateMachine.StateMachine.Instance.CurrentState is HostState)
+                AudioSystem.Instance.PlayRandomSoundEffectOfType(AudioCollectionTypes.PlayerJoined);
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs]: This is only called on the server. Maybe move it to `PlayerJoined`. That method runs on all clients.

**Review by lucasvdiepen — CHANGES_REQUESTED** [review]

**Review by lucasvdiepen — APPROVED** [review]

**Review by lucasvdiepen — APPROVED** [review]

**Pull request merged**